cli/main.rs

```diff
@@ -701,6 +701,10 @@ async fn run_command(
     return run_from_stdin(flags).await;
   }
 
+  if !flags.has_permission() && flags.has_permission_in_argv() {
+    log::warn!("Permission flags have likely been incorrectly set after the module name. To grant permissions, define them before the module name.")
```

Comment:
We should probably refer to this as the "script argument" and not "module name" since that's what the help text calls it.

```
USAGE:
    deno run [OPTIONS] <SCRIPT_ARG>...
```

Reply:
If running a command like `deno run main.js --allow-read` the error message could read something like:
```
Permission flags have likely been incorrectly set after the module name. 

To grant permissions, set them before the module name:
  deno run --allow-read main.js
```

Reply:
I feel like we shouldn't add more logic that could go wrong.

Reply:
I think it's worth it for the DX of the user - seems like it could be easily tested with integration test?

Reply:
I don't think it's worth the extra complexity. A user should hopefully know what "script argument" means.

Reply:
I agree. It's simple enough to change and makes the fix clearer.

Reply:
I really don't think we need more logic here. People will hopefully rarely hit this.

Reply:
@dsherret, are you understanding that Bartek is suggesting that the changes should dynamically change according to the incorrect flag that the user has entered? If so, I get where you're coming from, but I don't think that's what he's saying... Is that right, @bartlomieju?

Reply:
I meant dynamic error, but a static one seems fine too - I think it's much clearer for people not yet familiar with Deno's permission system.

Reply:
To clarify, I think this static text is the solution:
```
Permission flags have likely been incorrectly set after the script argument. 

To grant permissions, set them before the script argument:
    deno run [FLAG] main.js
```

Reply:
A static message sounds much better.

```
To grant permissions, set them before the script argument. For example:
    deno run --allow-read main.js
```

Reply:
> A static message sounds much better.
> 
> ```
> To grant permissions, set them before the script argument. For example:
>     deno run --allow-read main.js
> ```

Yeah, I like that.